ambry-rest/src/main/java/com.github.ambry.rest/NettyMessageProcessor.java

```diff
@@ -302,14 +306,12 @@ private void handleContent(HttpContent httpContent)
       long processingStartTime = System.currentTimeMillis();
       nettyMetrics.bytesReadRate.mark(httpContent.content().readableBytes());
       requestContentFullyReceived = httpContent instanceof LastHttpContent;
+      logger.trace("Received content for request - {}", request.getUri());
       try {
-        logger.trace("Received content for request - {}", request.getUri());
-        try {
-          request.addContent(httpContent);
-        } catch (IllegalStateException e) {
-          nettyMetrics.contentAdditionError.inc();
-          throw new RestServiceException(e, RestServiceErrorCode.InvalidRequestState);
-        }
+        request.addContent(httpContent);
+      } catch (IllegalStateException e) {
+        nettyMetrics.contentAdditionError.inc();
+        throw new RestServiceException(e, RestServiceErrorCode.InvalidRequestState);
```

Comment:
It eventually is because this exception triggers the `exceptionCaught()` function in the same class.

But I understand that two ways of doing the same thing is confusing - so I made this uniform and I don't throw exceptions anymore.